src/pages/posts/[slug].tsx

```diff
@@ -40,11 +40,14 @@ export async function unstable_getStaticProps({ params: { slug } }) {
 // Return our list of blog posts to prerender
 export async function unstable_getStaticPaths() {
   const postsTable = await getBlogIndex()
-  return Object.keys(postsTable).map(slug => getBlogLink(slug))
+  return {
+    paths: Object.keys(postsTable).map(slug => getBlogLink(slug)),
+  }
 }
 
 const RenderPost = ({ post, tweets, redirect }) => {
-  if (redirect) {
+  // XXX: why sometimes post is not here?
```

Comment:
@ijjk maybe you know why on some posts `post` is undefined here? 

I added logs to `unstable_getStaticProps` and it always works, and the posts that fail seem to be random 🤔

Reply:
In newer versions of Next.js we have added a fallback mode which is generated without passing any props to allow you to show a loading state for non-prerendered paths. In the latest canary this is required to be configured by setting `fallback: true/false` along with `paths` in `getStaticPaths`. 

If set to `false` any non-prerendered paths will 404 and when set to `true` it will render the page at build time with `useRouter().isFallback` set to true to signal that you should render the loading state

Reply:
hmm, and this cause post to be undefined at build time?

to debug, I tried passing along `slug` as well... all props were undefined when that happens... 🤔

Reply:
Yeah `post` will be `undefined` at build time when generating the fallback along with any other props. Basically if `useRouter().isFallback` is `true` then you shouldn't try to read any of the props and only render the loading state you want to show when a path is requested that wasn't prerendered at build time which can be as simple as `return null` if you don't want to render anything specific when the content is loading for the non-prerendered page